source/common/http/header_map_impl.h

```diff
@@ -63,6 +63,15 @@ class HeaderMapImpl : public virtual HeaderMap, NonCopyable {
   static void
   initFromInitList(HeaderMap& new_header_map,
                    const std::initializer_list<std::pair<LowerCaseString, std::string>>& values);
+  template <class It> static void initFromInitList(HeaderMap& new_header_map, It begin, It end) {
```

Comment:
This will need some kind of warning about passing an iterator to keys that are lower cased, otherwise things will break badly. In addition to a warning, add an ASSERT below that the key is lower cased.

Reply:
I think that we can reject not lowercased strings when compilation.